### PR TITLE
feat(fluid-tabs): added fluid tabs component

### LINKED ISSUE #325: Fluid tabs
### Description
Create the component as shown in the video attached below.

### Animation Preview
- Screenshot/Video:

https://github.com/user-attachments/assets/bf8fc0dc-f88b-4a28-a966-83915a827174

> [!NOTE]  
> Please refer to the **Animata contributing guidelines** available [here](https://www.animata.design/docs/contributing) for rules on how to contribute to the project. Let us know in the comments if you’re working on this issue, and feel free to ask any questions!

### Requirements
1. Create a new animated component that matches the provided design.
2. Add example usage of the component in the documentation or storybook.
3. **Add credits** for any resources, references, or assets used, as specified in the **Additional Resources** section.

<details>
  <summary>
    <h3> Guidelines & Best Practices</h3>
  </summary>

- Write reusable and clean code following the project’s style guide.
- The component should be **fully customizable**. The user should be able to modify props such as duration, easing, color, and size where applicable.
- Follow the [Contributing Guidelines](https://www.animata.design/docs/contributing) before submitting your pull request.

</details>

> [!IMPORTANT]  
> To ensure more contributors have the opportunity to participate, we kindly request that:
> - Each contributor submits only **one pull request** related to this issue.
> - If you're interested in working on this issue, please comment below. We will assign the issue on a **first-come, first-served basis**.

---

### Additional Resources
Interaction by [Nitish Khagwal](https://x.com/nitishkmrk)

---

> [!Tip]
> Join our Discord community if you need help or have any questions. We're here to help you get started and guide you through the process. [Join now](https://discord.gg/YfvqMf5MTE)

### PR DESCRIPTION
**Pull request by Rudra-Sankha-Sinhamahapatra**
Created the fluid tabs component
Fixes #325 

Hey @hari can you please review it

https://github.com/user-attachments/assets/d1e7b3db-12d0-4c42-80c6-a1499402b0b1

### CODE REVIEW

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@hari  can you please review this PR?

File: animata/card/fluid-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    timeoutRef.current = setTimeout(() => {
+      setTouchedTab(null);
+    }, 300); // Blur effect duration
+  };
+
+  const getTabIndex = (tabId: string) => tabs.findIndex((tab) => tab.id === tabId);
+
+  return (
+    <div className="flex h-screen items-center justify-center bg-gray-100">
```
Comment (hari) [inline animata/card/fluid-tabs.tsx:46]: Please remove `h-screen`

File: animata/card/fluid-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        </AnimatePresence>
+        {tabs.map((tab) => (
+          <motion.button
+            key={tab.id}
+            className={`relative z-10 rounded-full px-4 py-2 text-sm font-medium transition-colors duration-300 ${
+              activeTab === tab.id ? "text-black" : "text-gray-500"
+            } ${touchedTab === tab.id ? "blur-sm" : ""}`}
+            onClick={() => handleTabClick(tab.id)}
+            whileHover={{ scale: 1.05 }}
+            whileTap={{ scale: 0.95 }}
```
Comment (hari) [inline animata/card/fluid-tabs.tsx:66]: Please remove these two.

File: animata/card/fluid-tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          >
+            <span className="flex items-center space-x-2">
```
Comment (hari) [inline animata/card/fluid-tabs.tsx:68]: You can remove this span and use the children directly inside the button. The icon and span are not aligned vertically.

**Review by hari — CHANGES_REQUESTED** [review]

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
image
@hari done

File: animata/card/fluid-tabs.tsx (hunk 1 above)
Comment (Rudra-Sankha-Sinhamahapatra) [reply animata/card/fluid-tabs.tsx]: removed sir

File: animata/card/fluid-tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+import { useEffect, useRef, useState } from "react";
+import { AnimatePresence, motion } from "framer-motion";
+import { Inbox, Landmark, PieChart } from "lucide-react";
+
+const tabs = [
+  {
+    id: "accounts",
+    label: "Accounts",
+    icon: <Landmark size={18} className="mr-2 mt-1" />,
```
Comment (hari) [inline animata/card/fluid-tabs.tsx:11]: Please remove `mr-1 mt-1` from all three icons.

File: animata/card/fluid-tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            initial={{ x: `${getTabIndex(prevActiveTab) * 100}%` }}
+            animate={{ x: `${getTabIndex(activeTab) * 100}%` }}
+            transition={{ type: "spring", stiffness: 300, damping: 30 }}
+            style={{ width: `${100 / tabs.length}%` }}
+          />
+        </AnimatePresence>
+        {tabs.map((tab) => (
+          <motion.button
+            key={tab.id}
+            className={`relative z-10 flex w-full items-center justify-center px-5 py-3 text-sm font-bold transition-colors duration-300 ${
```
Comment (hari) [inline animata/card/fluid-tabs.tsx:70]: Add `gap-1.5` on the button className.

**Review by hari — CHANGES_REQUESTED** [review]
@Rudra-Sankha-Sinhamahapatra Maybe blurring the other items rather than the clicked one will look better? Please check and make the changes if it feels right.

File: animata/card/fluid-tabs.tsx (hunk 5 above)
Comment (Rudra-Sankha-Sinhamahapatra) [reply animata/card/fluid-tabs.tsx]: done

File: animata/card/fluid-tabs.tsx (hunk 4 above)
Comment (Rudra-Sankha-Sinhamahapatra) [reply animata/card/fluid-tabs.tsx]: done

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@hari sir i tried your suggestions but i think the current animation which was in the issue looks better, so i am currently not changing anything else ... but i have made those ui changes you mentioned..kindly check it please

**Comment by Rudra-Sankha-Sinhamahapatra** [comment]
@hari can you check now please

**Review by hari — APPROVED** [review]

**Pull request merged**